<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QxrdIntegrationWindow</class>
 <widget class="QMainWindow" name="QxrdIntegrationWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1487</width>
    <height>899</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_5">
    <item row="0" column="0">
     <widget class="QSplitter" name="m_Splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QTabWidget" name="m_Left">
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="tab_3">
        <attribute name="title">
         <string>Files</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout_4">
         <item row="0" column="0">
          <widget class="QxrdFileBrowserWidget" name="m_FileBrowserWidget" native="true"/>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="tab_4">
        <attribute name="title">
         <string>Data Items</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout_6">
         <item row="0" column="0">
          <widget class="QcepDatasetBrowserView" name="m_DatasetBrowserView">
           <property name="acceptDrops">
            <bool>true</bool>
           </property>
           <property name="dragEnabled">
            <bool>true</bool>
           </property>
           <property name="dragDropOverwriteMode">
            <bool>false</bool>
           </property>
           <property name="dragDropMode">
            <enum>QAbstractItemView::DragDrop</enum>
           </property>
           <property name="defaultDropAction">
            <enum>Qt::TargetMoveAction</enum>
           </property>
           <property name="alternatingRowColors">
            <bool>true</bool>
           </property>
           <property name="selectionMode">
            <enum>QAbstractItemView::ExtendedSelection</enum>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
      <widget class="QTabWidget" name="m_Middle">
       <property name="currentIndex">
        <number>1</number>
       </property>
       <widget class="QWidget" name="tab">
        <attribute name="title">
         <string>Image</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="0">
          <widget class="QxrdImagePlotWidget" name="widget" native="true"/>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="tab_2">
        <attribute name="title">
         <string>Integrated</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout_7">
         <item row="0" column="0">
          <widget class="QxrdIntegratedPlotWidget" name="widget_2" native="true"/>
         </item>
        </layout>
       </widget>
      </widget>
      <widget class="QTabWidget" name="m_Right">
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="tab_5">
        <attribute name="title">
         <string>Integration</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout">
         <item row="2" column="0">
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QPushButton" name="m_ClearGraphButton">
             <property name="toolTip">
              <string>Clear all curves from integrated data graph</string>
             </property>
             <property name="text">
              <string>Clear All Curves</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="m_IntegrateButton">
             <property name="text">
              <string>Integrate</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="m_ClearSelectedGraphButton">
             <property name="toolTip">
              <string>Clear selected curves from integrated data graph
(Click legend entries to select curves)</string>
             </property>
             <property name="text">
              <string>Clear Selected Curves</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="m_IntegrateOptionsButton">
             <property name="toolTip">
              <string>More intergration options</string>
             </property>
             <property name="text">
              <string>More Options...</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="1" column="0">
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>498</width>
             <height>367</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="0" column="0">
          <layout class="QFormLayout" name="formLayout">
           <item row="0" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Oversampling Factor</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QSpinBox" name="m_OversampleFactor">
             <property name="toolTip">
              <string>Specify oversampling factor for angular integration</string>
             </property>
             <property name="specialValueText">
              <string>No oversampling</string>
             </property>
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>16</number>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Integrate vs</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QComboBox" name="m_IntegratorXUnits">
             <property name="toolTip">
              <string>Choose X units for integration</string>
             </property>
             <item>
              <property name="text">
               <string>Integrate vs 2 Theta</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Integrate vs Q</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Integrate vs R</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Integration Step Size</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QxrdCenterStepSpinner" name="m_IntegratorStepSize">
             <property name="toolTip">
              <string>step size for integration
if ==  0, use fixed number of steps instead</string>
             </property>
             <property name="specialValueText">
              <string>Use fixed number of steps</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>Integration Number of Steps</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QSpinBox" name="m_IntegratorNSteps">
             <property name="toolTip">
              <string>Number of steps for integration
if == 0, use step size instead</string>
             </property>
             <property name="specialValueText">
              <string>Use fixed step size</string>
             </property>
             <property name="maximum">
              <number>999999</number>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Integration Minimum</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QDoubleSpinBox" name="m_IntegratorMinimum">
             <property name="toolTip">
              <string>Start value for integration</string>
             </property>
             <property name="decimals">
              <number>5</number>
             </property>
             <property name="maximum">
              <double>10000.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="5" column="0">
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>Integration Maximum</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="5" column="1">
            <widget class="QDoubleSpinBox" name="m_IntegratorMaximum">
             <property name="toolTip">
              <string>Ending value for integration range</string>
             </property>
             <property name="decimals">
              <number>5</number>
             </property>
             <property name="maximum">
              <double>10000.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="6" column="0">
            <widget class="QCheckBox" name="m_EnableSelfNormalization">
             <property name="toolTip">
              <string>Normalize integrated curves so that the average value in the self normalization range is 1</string>
             </property>
             <property name="text">
              <string>Enable Self Normalization </string>
             </property>
            </widget>
           </item>
           <item row="7" column="0">
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>Self Normalization Minimum</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="7" column="1">
            <widget class="QDoubleSpinBox" name="m_SelfNormalizationMinimum">
             <property name="toolTip">
              <string>Start value for self normalization range</string>
             </property>
             <property name="decimals">
              <number>5</number>
             </property>
             <property name="maximum">
              <double>10000.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="8" column="0">
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>Self Normalization Maximum</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="8" column="1">
            <widget class="QDoubleSpinBox" name="m_SelfNormalizationMaximum">
             <property name="toolTip">
              <string>End value for self normalization</string>
             </property>
             <property name="decimals">
              <number>5</number>
             </property>
             <property name="maximum">
              <double>10000.000000000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="tab_6">
        <attribute name="title">
         <string>Corrections</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="0" column="0">
          <layout class="QFormLayout" name="formLayout_2">
           <item row="0" column="0" colspan="2">
            <widget class="QCheckBox" name="m_EnableGeometry">
             <property name="toolTip">
              <string>Enable geometry corrections during integration
intensity *= pow(distance/m_DetectorDistance, 2)/cos(tth);</string>
             </property>
             <property name="text">
              <string>Enable Geometry Corrections </string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_30">
             <property name="text">
              <string>User Geometry Kind</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QComboBox" name="m_EnableUserGeometry">
             <item>
              <property name="text">
               <string>None</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>User Defined Geometry</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_25">
             <property name="text">
              <string>User Geometry Func</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLineEdit" name="m_UserGeometryFunction">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="label_27">
             <property name="text">
              <string>User Geometry Script</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QPushButton" name="m_UserGeometryEdit">
             <property name="text">
              <string>Edit...</string>
             </property>
            </widget>
           </item>
           <item row="4" column="0" colspan="2">
            <widget class="QCheckBox" name="m_EnablePolarization">
             <property name="toolTip">
              <string>Enable polarization correction during integration:
intensity *= (m_Polarization      *(1 - pow(sin(chi)*sin(tth),2)) +
                   (1 - m_Polarization)*(1 - pow(cos(chi)*sin(tth),2)));</string>
             </property>
             <property name="text">
              <string>Enable Polarization Corrections </string>
             </property>
            </widget>
           </item>
           <item row="5" column="0">
            <widget class="QLabel" name="label_24">
             <property name="text">
              <string>Polarization Factor</string>
             </property>
            </widget>
           </item>
           <item row="5" column="1">
            <widget class="QDoubleSpinBox" name="m_Polarization">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="toolTip">
              <string>Polarization factor : [0..1]</string>
             </property>
             <property name="keyboardTracking">
              <bool>false</bool>
             </property>
             <property name="prefix">
              <string/>
             </property>
             <property name="decimals">
              <number>6</number>
             </property>
             <property name="maximum">
              <double>1.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.010000000000000</double>
             </property>
            </widget>
           </item>
           <item row="6" column="0" colspan="2">
            <widget class="QCheckBox" name="m_EnableAbsorption">
             <property name="toolTip">
              <string>Enable absorption correction during integration
intensity *= 
  exp((distance-m_DetectorDistance)/m_AttenuationLength);</string>
             </property>
             <property name="text">
              <string>Enable Absorption Corrections </string>
             </property>
            </widget>
           </item>
           <item row="7" column="0">
            <widget class="QLabel" name="label_23">
             <property name="text">
              <string>Attenuation Length</string>
             </property>
            </widget>
           </item>
           <item row="7" column="1">
            <widget class="QDoubleSpinBox" name="m_AttenuationLength">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="toolTip">
              <string>Attenuation Length (mm)</string>
             </property>
             <property name="keyboardTracking">
              <bool>false</bool>
             </property>
             <property name="decimals">
              <number>6</number>
             </property>
             <property name="maximum">
              <double>1000000.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.010000000000000</double>
             </property>
             <property name="value">
              <double>1000.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="8" column="0">
            <widget class="QLabel" name="label_29">
             <property name="text">
              <string>User Absorption Kind</string>
             </property>
            </widget>
           </item>
           <item row="8" column="1">
            <widget class="QComboBox" name="m_EnableUserAbsorption">
             <item>
              <property name="text">
               <string>None</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>A function of pixel coordinates</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>A function of x,y displacement from beam center</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>A function of r,chi</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>A function of q,chi</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="9" column="0">
            <widget class="QLabel" name="label_26">
             <property name="text">
              <string>User Absorption Func</string>
             </property>
             <property name="wordWrap">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item row="9" column="1">
            <widget class="QLineEdit" name="m_UserAbsorptionFunction">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
            </widget>
           </item>
           <item row="10" column="0">
            <widget class="QLabel" name="label_28">
             <property name="text">
              <string>User Absorption Script</string>
             </property>
            </widget>
           </item>
           <item row="10" column="1">
            <widget class="QPushButton" name="m_UserAbsorptionEdit">
             <property name="text">
              <string>Edit...</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="1" column="0">
          <spacer name="verticalSpacer_2">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>388</width>
             <height>418</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1487</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="m_FileMenu">
    <property name="title">
     <string>Fi&amp;le</string>
    </property>
   </widget>
   <widget class="QMenu" name="m_EditMenu">
    <property name="title">
     <string>Edit</string>
    </property>
   </widget>
   <widget class="QMenu" name="m_WindowMenu">
    <property name="title">
     <string>W&amp;indow</string>
    </property>
    <addaction name="actionWindow"/>
   </widget>
   <addaction name="m_FileMenu"/>
   <addaction name="m_EditMenu"/>
   <addaction name="m_WindowMenu"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionWindow">
   <property name="text">
    <string>&amp;Window</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QcepDatasetBrowserView</class>
   <extends>QTreeView</extends>
   <header>qcepdatasetbrowserview.h</header>
  </customwidget>
  <customwidget>
   <class>QxrdFileBrowserWidget</class>
   <extends>QWidget</extends>
   <header>qxrdfilebrowserwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QxrdCenterStepSpinner</class>
   <extends>QDoubleSpinBox</extends>
   <header>qxrdcenterstepspinner.h</header>
  </customwidget>
  <customwidget>
   <class>QxrdImagePlotWidget</class>
   <extends>QWidget</extends>
   <header>qxrdimageplotwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QxrdIntegratedPlotWidget</class>
   <extends>QWidget</extends>
   <header>qxrdintegratedplotwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
